### PR TITLE
Generate habitatmap_terr 2020 v2 with unique PK

### PR DESCRIPTION
**Pull request by cecileherr**
- created habitatmap_terr_2020_v2
- updated the renv (based on a clean update of all the packages in april 2024)
- removed the duplicate in the primary key based on {polygon_id + type + certain} (see [issue 66](https://github.com/inbo/n2khab-preprocessing/issues/66))

_(all the other updates mentioned in  [PR #65 ](https://github.com/inbo/n2khab-preprocessing/pull/65#issuecomment-1996880418), will be for the next version based on habitatmap 2023)_

Bookdown: 
[generating_habitatmap_terr.zip](https://github.com/inbo/n2khab-preprocessing/files/15291483/generating_habitatmap_terr.zip)

**A few side notes:**
- code based on branch  `update_habitatmap_stdized_habitatmap_terr` (based on habitatmap 2023) and trying to merge to main. But this adds many commits that are not relevant to this PR (I only worked on habitatmap_terr). Suggestions to do this cleanly are welcome.
- aggregation of the PK: I chose to keep track of the detailed sources that is:

2020 v1
```
#>   polygon_id   type      certain code_orig  phab source                         
#>   <fct>        <fct>     <lgl>   <chr>     <int> <fct>                          
#> 1 427121_v2020 7140_meso TRUE    7140         95 habitatmap_stdized + interpret…
#> 2 427121_v2020 7140_meso TRUE    7140_meso     5 habitatmap_stdized
```
becomes this in 2020 v2:

```
#>   polygon_id   type      certain code_orig  phab source                         
#>   <fct>        <fct>     <lgl>   <chr>     <int> <fct>                          
#> 1 427121_v2020 7140_meso TRUE    7140;7140_meso         100 habitatmap_stdized + interpretation;habitatmap_stdized
```
Let me know if you would prefer the simpler version:

```
#>   polygon_id   type      certain code_orig  phab source                         
#>   <fct>        <fct>     <lgl>   <chr>     <int> <fct>                          
#> 1 427121_v2020 7140_meso TRUE    7140;7140_meso         100 habitatmap_stdized + interpretation
```
- note 2020_**v2** for habitatmap_terr is based on 2020_**v1** of habitatmap_stdized. I suppose it is not a problem
- tested: `read_habitatmap_terr() `works as expected, the differences between 2020_v1 and v2 are as expected (limited to `polygon_id` = 427121_v2020 and one more level for the sources in v2)
- I will modify `read_habitatmap_terr()` in `n2khab` to add this version to the list of versions
```
read_habitatmap_terr(
     file = file.path(locate_n2khab_data(),
                      "20_processed/habitatmap_terr/habitatmap_terr.gpkg"),
     keep_aq_types = TRUE,
     drop_7220 = TRUE,
     version = c("habitatmap_terr_2020_v2", "habitatmap_terr_2020_v1", 
"habitatmap_terr_2018_v2",  "habitatmap_terr_2018_v1")
 )
```

### CODE REVIEW

**Comment by florisvdh** [comment]
> But this adds many commits that are not relevant to this PR (I only worked on habitatmap_terr). Suggestions to do this cleanly are welcome.

- Since PR [#65](https://github.com/inbo/n2khab-preprocessing/pull/65) is hybrid, i.e. tackling several directories at once, that may indeed be problematic in getting a focused PR here.
- You would (probably) like to get into a state in which this branch (hence PR) only contains the commits you'd like to merge into `main` now, regarding `hmt_2020v2`? (And that may include _some_ of the commits from PR 65.) The solution will be to either:
  - refactor this branch with `git rebase -i main` (being on the `hms_hmt_2020v2` branch), setting unwanted commits as 'drop' in the `git-rebase-todo` list and then saving it. After that the rebase will be performed: unwanted commits will be kicked out. You _may_ encounter conflicts along the way, especially if a single commit is a mix of several things (then best edit that commit and amend it), but `git rebase` will tell you how to proceed the rebase after fixing them. You need a clean working dir to start (`git stash` may be of help). Before doing the rebase, make sure to backup current branch state with e.g. `git branch mybackupbranch`. After the rebase succeeds and the result is indeed as you want, force push using `git push --force`.
  - backup current branch state with e.g. `git branch mybackupbranch`, then reset `hms_hmt_2020v2` to `main` using `git reset --hard main` and build that branch up again by picking only the commits that you want in, using `git cherry-pick <commithash>`, one by one (i.e. commits from both `update_habitatmap_stdized_habitatmap_terr` and `mybackupbranch` branches). And then force-push as above.

I hope to get at reviewing later today; coming days may be too busy.

**Comment by cecileherr** [comment]
Thx! I will now try to drop the commits related to generate_habitatmap_stdized and generate_watersurfaces_hab (=> limit this PR to generate_habitatmap_terr). You can expect an update in the next hour or so

**Comment by cecileherr** [comment]
> Thx! I will now try to drop the commits related to generate_habitatmap_stdized and generate_watersurfaces_hab (=> limit this PR to generate_habitatmap_terr). You can expect an update in the next hour or so

update done

File: src/generate_habitatmap_terr/20_generate_habmap_terr_interpreted.Rmd [hunk 1]
```diff
@@ -262,10 +262,10 @@ habmap_polygons_water <-
     filter(opp == oppsum)
 
 habmap_polygons_interpreted <- habmap_polygons_interpreted %>% 
-    anti_join(habmap_polygons_water)
+    anti_join(habmap_polygons_water, by = join_by(polygon_id))
 
 habmap_types_interpreted <- habmap_types_interpreted %>% 
-    anti_join(habmap_polygons_water)    
+    anti_join(habmap_polygons_water, by = join_by(polygon_id))
```
Comment (florisvdh) [inline src/generate_habitatmap_terr/20_generate_habmap_terr_interpreted.Rmd:268]: Good to always be explicit about the key pair indeed.

In this case I presume (hope) this did not affect results?

**Review by florisvdh — APPROVED** [review]
Thanks for cleaning the branch!

I tried but failed to find mistakes.

One extra suggestion could be to replace the checksum generation in `30_check_result.Rmd` by:

https://github.com/inbo/n2khab-preprocessing/blob/0b94cd25a76f77bd802dbc65cd4ebe3fb7fb4a5e/src/update_shallowgroundwater/20_check_result.Rmd#L10-L13

Also, can you re-run this bookdown project (if not done already) and upload the compiled HTML again; the one that you posted first still contains changes (at least text) that you didn't include in the rewritten branch. Of course, take care of the produced data source version since it will need to end up in Zenodo (maybe on par with {n2khab} merges).

> aggregation of the PK: I chose to keep track of the detailed sources

OK!

> 2020_v2 for habitatmap_terr is based on 2020_v1 of habitatmap_stdized

:+1: 

> I will modify `read_habitatmap_terr()` in `n2khab`

Excellent.

**Comment by cecileherr** [comment]
Updated version bookdown
[generating_habitatmap_terr.zip](https://github.com/inbo/n2khab-preprocessing/files/15307091/generating_habitatmap_terr.zip)

**Pull request merged**